daemon/api.go

```diff
@@ -2137,6 +2137,22 @@ func runSnapctl(c *Command, r *http.Request, user *auth.UserState) Response {
 	context, _ := c.d.overlord.HookManager().Context(snapctlOptions.ContextID)
 	stdout, stderr, err := ctlcmdRun(context, snapctlOptions.Args, uid)
 	if err != nil {
+		if e, ok := err.(*ctlcmd.UnsuccessfulError); ok {
+			result := map[string]interface{}{
+				"stdout":    string(stdout),
+				"stderr":    string(stderr),
+				"exit-code": fmt.Sprintf("%d", e.ExitCode), // string to avoid any int/float64 trickery
```

Comment:
I would really prefer we passed this as a number, and dealt with the issues in the client